tests/gunicorn_utils.py

```diff
@@ -77,4 +84,14 @@ def run_gunicorn(config_path: str = "config/gunicorn.conf.py", **kwargs) -> Iter
 
         yield
     finally:
+        # See https://github.com/python-discord/snekbox/issues/177
+        # Sleeping before terminating the process avoids a case where
+        # terminating the process can take >30 seconds.
+        time.sleep(0.2)
+
         proc.terminate()
+
+        # Actually wait for the process to finish. There doesn't seem to be a
+        # reliable way of checking if the timeout was reached or this ended normally,
+        # but if the timeout is reached it will probably error later anyway.
+        proc.join(timeout=10)
```

Comment:
Can try to kill the process afterwards in case it timed out? So ultimately, we are first trying to gracefully stop it with `terminate`, and if that fails, we `kill` it.

Reply:
Done in https://github.com/python-discord/snekbox/pull/180/commits/3a10f404064eb520d5f6ac882dc5e9429654ef8b